packages/pwa-kit-runtime/src/utils/ssr-server/hybridProxy.js

```diff
@@ -0,0 +1,161 @@
+                            proxyRes?.headers?.location.includes(sfccOrigin)
+                        ) {
+                            logger.info(`Rewriting location header => ${proxyRes.headers.location}`)
+                            res.setHeader(
+                                'location',
+                                proxyRes.headers.location.replace(sfccOrigin, proxyOrigin)
+                            )
+                        }
+
+                        // prefix /on/demandware.static/ with /mobify/proxy/dwrestatic to force
+                        // static assets through an MRT proxy; there should be no reason
+                        // to process this in runtime code (nor should there be POST requests for these)
+                        // requires a unique MRT proxy configure; ignore DIS CDN links that look similar
+                        updatedResponse = updatedResponse.replace(
+                            new RegExp(`(?<!dw/image/.+?/.+?)/on/demandware\\.static/`, 'g'),
+                            `/mobify/proxy/dwrestatic/on/demandware.static/`
+                        )
+                        return updatedResponse
+                    case 'application/json':
+                        try {
+                            response = JSON.parse(responseBuffer.toString('utf8'))
+                            return JSON.stringify(
+                                iterate(response, null, {sfccOrigin, proxyOrigin})
+                            )
+                        } catch (e) {
+                            logger.error(`error parsing JSON input: ${e}`)
+                            return responseBuffer
+                        }
+                    default:
+                        return responseBuffer
+                }
+            })(proxyRes, req, res)
+        }
+    }
+
+    // Attach the proxy middleware if we are in hybrid mode
+    return createProxyMiddleware(function (pathname, req) {
+        let match = hybridRoutingRules.some((rule) =>
+            evaluateRule(rule, {
+                host: req.hostname,
+                uri: req.url,
+                path: pathname,
+                cookies: req.headers.cookie || ''
+            })
+        )
+
+        // hybridRoutingRules(MRT eCDN rules) are evaluated to determine what gets sent to MRT
+        // https://developer.salesforce.com/docs/commerce/commerce-api/references/cdn-api-process-apis?meta=createMrtRules
+        // So the traffic we proxy to SFCC will be the opposite
+        return !match
+    }, proxyOptions)
+}
+
+const isString = (element) => typeof element === 'string'
+const isArray = (element) => Array.isArray(element)
+const isObject = (element) => element && typeof element === 'object'
+const isIterable = (element) => isArray(element) || isObject(element)
+
+const forEachIn = (iterable, functionRef) => {
+    Object.keys(iterable).forEach((key) => {
+        functionRef(key, iterable[key])
+    })
+}
```

Comment:
These appear to be generic util functions. Do they need to be a part of hybridProxy.js ?